chapters/image_accceses.md

```diff
@@ -0,0 +1,146 @@
+
+This is possible because the handle to the image is read-only, but the underlying texel data of the image is mutable.
+
+A software mental model of the `OpTypeImage` is to think of it as container that holds metadata while also holding the data of the image.
+
+```c++
+struct ImageType {
+  bool depth;
+  bool arrayed;
+  bool sampled;
+  bool multisampled;
+  Format format;
+  int width;
+  // other metadata ...
+  void* texels; // data of the image (which is mutable)
+}
+```
+
+The above `OpVariable` is read-only because we are not allowed to adjust its format, width, or multisampling state ever.
+
+# Sampled Images
+
+When using a sampler, you must have an `OpTypeSampledImage` object, this will be directly pointing to an image object of `OpTypeImage`.
+
+To access the sampled image you will use an `OpImage*Sample*` instruction (ex. `OpImageSampleImplicitLod`) to get an `OpSampledImage`. (It can also use an `OpImage*Gather` as well)
+
+![image_access_sampled_image.png](../images/image_access_sampled_image.png)
+
+## Fetch
+
+There are also `OpImageFetch` (and `OpImageSparseFetch`) instructions which work similar as `OpImage*Sample*`.
+
+The `OpImageFetch` only takes integer texel coordinates. The texture data is read directly without any sampling operations.
+
+This means an `OpImageFetch` will directly access through an `OpImage` and doesn't need an `OpSampledImage` object.
+
+# Non-sampled image
+
+When not using a sampler (ex. storage image), you can directly use an loaded image object of `OpTypeImage`.
+
+Instructions such as `OpImageRead` and `OpImageWrite` will use the result of a `OpLoad`.
+This `OpLoad` will be loading in a variable of type image (`OpTypeImage`).
+
+![image_access_storage_image.svg](../images/image_access_storage_image.svg)
+
+## Loading handle vs loading texel
+
+The following code/diagram shows an example how an `OpImageWrite` instruction is used.
+
+![image_access_handle.svg](../images/image_access_handle.svg)
+
+From the code, we se that the `OpImageWrite` calls `OpLoad` to load the the handle (`image_view_a`/`image_view_b`).
+From there is then does a "write" to update the texel data.
+The "write" did not update the handle/pointer, but the texel data itself.
+This means there was no "read access" texel data.
+
+SPIR-V has decorations for `NonReadable`/`NonWritable` that can be applied to an image as well.
+These decorations are applied to the texel data, not to the handles.
+
+# OpImage
+
+With many other things also starting with `OpImage*` it is easy to lose track what `OpImage` is for. As the spec says simply:
+
+>  [OpImage] Extract the image from a sampled image.
+
+The `OpImage` instruction extracts image object of `OpTypeImage` from an `OpTypeSampledImage`.
+The `OpImage` is describing an action, rather than the input to or the output from an action.
+
+This means you will **not** be using an `OpImage` when accessing an image without a sampler.
```

Comment:
This is perhaps slightly oversimplified.  If you are starting from an object of type `OpTypeImage` then you already have what you need to access an image without a sampler.  If you are starting with an object of type `OpTypeSampledImage` though, you need a way to extract the object of type `OpTypeImage` from it, in which case you will need to use `OpImage`.

Maybe:

```suggestion
The `OpImage` instruction is useful if you have an object of type `OpTypeSampledImage` and you need an object of type `OpTypeImage`, say to access the image data without a sampler.
```